gulpfile.js

```diff
@@ -110,6 +113,45 @@ gulp.task('test:watch', 'Run all tests in PhantomJS and watch', (done) => {
 });
 // </editor-fold>
 
+gulp.task('dependencies:docs', () => {
+    return gulp.src([
+        'node_modules/codemirror/lib/codemirror.js',
+        'node_modules/jquery/dist/jquery.js',
+        'node_modules/redux/dist/redux.js',
+        'node_modules/react/dist/react.js',
+        'node_modules/react-dom/dist/react-dom.js',
+        'node_modules/react-redux/dist/react-redux.js',
+        'node_modules/react-bootstrap/dist/react-bootstrap.js',
+        'node_modules/rc-slider/dist/rc-slider.js',
+        'node_modules/underscore/underscore.js',
+        'node_modules/moment/min/moment-with-locales.js',
+        'node_modules/coveo-styleguide/dist/js/VaporSVG.js',
+    ])
+        .pipe(concat('react-vapor.dependencies.js'))
+        .pipe(gulp.dest('./docs/dependencies/'));
+});
+
+gulp.task('dependencies:prod', () => {
+    return gulp.src([
+        'node_modules/codemirror/lib/codemirror.js',
+        'node_modules/jquery/dist/jquery.min.js',
+        'node_modules/redux/dist/redux.min.js',
+        'node_modules/react/dist/react.min.js',
+        'node_modules/react-dom/dist/react-dom.min.js',
+        'node_modules/react-redux/dist/react-redux.min.js',
+        'node_modules/react-bootstrap/dist/react-bootstrap.min.js',
+        'node_modules/rc-slider/dist/rc-slider.min.js',
+        'node_modules/underscore/underscore-min.js',
+        'node_modules/moment/min/moment-with-locales.min.js',
+        'node_modules/coveo-styleguide/dist/js/VaporSVG.js',
+    ])
+        .pipe(gulp.dest('./dist/dependencies/'))
+        .pipe(concat('react-vapor.dependencies.js'))
+        .pipe(gulp.dest('./dist/'));
+});
+
+gulp.task('dependencies', ['dependencies:prod', 'dependencies:docs']);
+

```

Comment:
Can we do all of this with webpack instead?

Reply:
We can probably add an entry point in the webpack config and automatically put everything that comes from node_modules in another chunk. It will handle the dev and prod version automatically

See https://stackoverflow.com/a/49429500 for inspiration

Reply:
gulp is really good for stream processing, since we're only taking a bunch of files, concatenate them and move them to another places, i think gulp is really well suited for the job.  Can you specify how you would see this implemented in webpack?  
  
On a related topic, one thing I had in mind though is to provide a file containing the proper externals setup in webpack. Meaning we could have a js file, say `react-vapor.webpackexternals.js` :  
  
```  
module.exports = {
        codemirror: 'CodeMirror',
        jquery: '$',
        react: 'React',
        'react-dom': 'ReactDOM',
        'react-bootstrap': 'ReactBootstrap',
        'react-redux': 'ReactRedux',
        Slider: {root: ['rc-slider', 'Slider']},
        createSliderWithTooltip: {root: ['rc-slider', 'createSliderWithTooltip']},
        'coveo-styleguide': 'VaporSVG',
        redux: 'Redux',
        underscore: '_',
        moment: 'moment',
}  
```  
  
This would then be really simple for react-vapor users having webpack as their bundler to setup react-vapor properly in their build. They would just have to load `dist/react-vapor.dependencies.js` in the browser and to destructure this file under `externals` in their webpack.config and they would be good to go after that.

Reply:
@GermainBergeron will look into it. Feel free to comment on my comment as well.

Reply:
It was more of a suggestion than anything. I think it would work the way @GermainBergeron said, but if it is too complicated, it can be made some other time as well.

Reply:
not sure @GermainBergeron 's proposition is the best strategy after reading this https://webpack.js.org/guides/author-libraries/#src/components/Sidebar/Sidebar.jsx  , will continue tomorrow